security/src/main/java/com/fincity/security/dao/UserDAO.java

```diff
@@ -409,4 +414,114 @@ public Mono<Set<ULong>> getUserListFromClientIds(Set<ULong> clientList) {
 
 		return Mono.just(users);
 	}
+
+	public Mono<Boolean> checkPasswordEqual(ULong userId, String newPassword) {
+
+		return Mono.from(
+
+		        this.dslContext.select(SECURITY_USER.PASSWORD)
+		                .from(SECURITY_USER)
+		                .where(SECURITY_USER.ID.eq(userId))
+		                .limit(1))
+		        .map(Record1::value1)
+		        .map(pass -> pass.equals(newPassword));
+
+	}
+
+	public Mono<Set<String>> getPastPasswords(ULong userId) {
+
+		return Flux.from(
+
+		        this.dslContext.select(SECURITY_PAST_PASSWORDS.PASSWORD)
+		                .from(SECURITY_PAST_PASSWORDS)
+		                .where(SECURITY_PAST_PASSWORDS.USER_ID.eq(userId)))
+		        .map(Record1::value1)
+		        .collectList()
+		        .map(HashSet::new);
+	}
+
+	public Mono<Boolean> updatePassword(ULong userId, String newPassword) {
+
```

Comment:
You need to encrypt the password and add it to the past passwords simultaneously.